jupyterlab_gcedetails/src/index.ts

```diff
@@ -23,19 +23,72 @@ import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin,
 } from '@jupyterlab/application';
+import {
+  getMetadata,
+  ClientTransportService,
+  ServerProxyTransportService,
+} from 'gcp_jupyterlab_shared';
+import { NotebooksService } from './service/notebooks_service';
+import { extractLast } from './data/data';
+import { DetailsService } from './service/details_service';
+import { PriceService } from './service/price_service';
+import { ServerWrapper } from './components/server_wrapper';
+
+async function activateDetailsWidget(
+  app: JupyterFrontEnd,
+  statusbar: IStatusBar
+) {
+  console.debug('Activating GCP Details Extension');
+
+  const clientTransportService = new ClientTransportService();
+  const serverProxyTransportService = new ServerProxyTransportService();
+  const detailsUrl = `gcp/v1/details`;
```

Comment:
I would just inline this since it's not used anywhere else or potentially even hide it inside of the ServerWrapper class itself.

Reply:
I've made the changes suggested, should be good to merge, just wanted your feedback first that's why I had it as WIP :)